contracts/cryptoblades.sol

```diff
@@ -436,6 +436,15 @@ contract CryptoBlades is Initializable, AccessControlUpgradeable {
         weapons.burn(burnID);
     }
 
+    function burnWeapons(uint256[] memory burnIDs) public
+            doesNotHaveMoreThanMaxCharacters
+            isWeaponsOwner(burnIDs) requestPayFromPlayer(ABDKMath64x64.fromUInt(burnWeaponFee.mulu(burnIDs.length))) {
+        _payContract(msg.sender, ABDKMath64x64.fromUInt(burnWeaponFee.mulu(burnIDs.length)));
```

Comment:
I don't think `ABDKMath64x64.fromUInt(burnWeaponFee.mulu(burnIDs.length))` will have the desired effect (for example, `3.5 * 3 = 10.5`, which would get rounded off to `10`). Instead I recommend `burnWeaponFee.mul(ABDKMath64x64.fromUInt(burnIDs.length))`.

```suggestion
            isWeaponsOwner(burnIDs) requestPayFromPlayer(burnWeaponFee.mul(ABDKMath64x64.fromUInt(burnIDs.length))) {
        _payContract(msg.sender, burnWeaponFee.mul(ABDKMath64x64.fromUInt(burnIDs.length)));
```

Reply:
> burnWeaponFee.mul(ABDKMath64x64.fromUInt

Great catch.  Thanks!